services/src/main/java/org/keycloak/services/resources/admin/GroupResource.java

```diff
@@ -150,6 +153,21 @@ public void deleteGroup() {
         adminEvent.operation(OperationType.DELETE).resourcePath(session.getContext().getUri()).success();
     }
 
+    @GET
+    @Path("children")
+    @NoCache
+    @Produces(MediaType.APPLICATION_JSON)
+    @Tag(name = KeycloakOpenAPI.Admin.Tags.GROUPS)
+    @Operation( summary = "Return a paginated list of subgroups that have a parent group corresponding to the group on the URL")
+    public Stream<GroupRepresentation> getSubGroups(@QueryParam("first") @DefaultValue("0") Integer first,
+        @QueryParam("max") @DefaultValue("10") Integer max,
+        @QueryParam("full") @DefaultValue("false") Boolean full) {
+        this.auth.groups().requireView(group);
+        return session.groups().searchForSubgroupsByParentIdStream(realm, group.getId(), first, max)
+            .filter(g -> auth.groups().canView() || auth.groups().canView(g))
```

Comment:
Two comments:

- I think the result of `auth.groups().canView()` can be cached somehow, so it is not needed to call for every group? This is related to my other comment in class `GroupsResource`

- I think that when `session.groups().searchForSubgroupsByParentIdStream(realm, group.getId(), first, max)` returns 10 groups (assuming that first is 0 and max is 10 for example), but then for instance 3 groups are filtered by admin fine-grained permission calls to `auth.groups().canView(g)`, then this method would return only 7 groups instead of 10.
This is bit misleading IMO, however I don't know if there is good solution for this, which will allow to return proper amount of groups and scale at the same time. Anyway, this problem seems to exist for all other objects using fine-grained permissions, so it is not a blocker for this PR IMO. And admin fine-grained permissions are preview feature, so we may need to discuss if/how to address this when we want to support it.

Reply:
Thanks for updating per my first comment. I don't think we can do something with my 2nd comment, so I am resolving this discussion as it is OK for me for this PR